apps/angular/src/app/features/anime/components/anime-table/anime-table.component.ts

```diff
@@ -11,23 +65,144 @@ import { AnimeService } from '../../../../../core/services/anime.service';
+  private pagination$!: BehaviorSubject<PaginationUrlParams>;
 
-  public constructor(private readonly animeService: AnimeService) {
-    this.animeList$ = animeService.getAnimeList();
+  /** Emits on sorting change. */
+  private sorting$!: BehaviorSubject<Sort>;
+
+  /** Emits on filtering change. */
+  private filtering$!: BehaviorSubject<readonly string[]>;
+
+  /** Emits on searching change. */
+  private search$!: BehaviorSubject<string>;
+
+  /** Search input value. */
+  public searchString!: string;
+
+  /** Subscription on changes on page. */
+  private subscriptionOnChanges = new Subscription();
+
+  /** Pagination component. */
+  @ViewChild(PaginationComponent)
+  private paginationComponent!: PaginationComponent;
+
+  /** Filtering component. */
+  @ViewChild(FilteringComponent)
+  private filteringComponent!: FilteringComponent;
+
+  public constructor(
+    private readonly animeService: AnimeService,
+    private readonly router: Router,
+    private readonly route: ActivatedRoute,
+  ) {
+    this.urlParams = this.route.snapshot.queryParams;
+    this.initializeTableInterface();
+  }
+
+  /** Initialize table interface. */
+  private initializeTableInterface(): void {
+    this.pagination$ = new BehaviorSubject<PaginationUrlParams>({
+      limit: this.urlParams[URL_PARAMS.limit] ?
+        this.urlParams[URL_PARAMS.limit] :
+        URL_DEFAULT_PARAMS.limit,
+      offset: this.urlParams[URL_PARAMS.offset] ?
+        this.urlParams[URL_PARAMS.offset] :
+        URL_DEFAULT_PARAMS.offset,
+    });
+    this.sorting$ = new BehaviorSubject<Sort>({
+      active: this.urlParams[URL_PARAMS.sort] ?
+        this.urlParams[URL_PARAMS.sort] :
+        URL_DEFAULT_PARAMS.sort,
+      direction: '',
+    });
+    this.filtering$ = new BehaviorSubject<readonly string[]>(
+      this.urlParams[URL_PARAMS.filter] ?
+        this.urlParams[URL_PARAMS.filter].split(',') :
+        URL_DEFAULT_PARAMS.filter,
+    );
+    this.search$ = new BehaviorSubject<string>(
+      this.urlParams[URL_PARAMS.search] ?
+        this.urlParams[URL_PARAMS.search] :
+        URL_DEFAULT_PARAMS.search,
+    );
+    this.searchString = this.urlParams[URL_PARAMS.search] ?
+      this.urlParams[URL_PARAMS.search] :
+      '';
+  }
+
+  /**
+   * Subscribes to change in url parameters.
+   * Creates new parameters for the http request and get anime list.
+   */
+  public getAnimeList(): void {
+    this.animeList$ = this.route.queryParamMap.pipe(
+      switchMap(params => {
+        params.keys.map(key => {
+          this.urlParams[key] = params.get(key) as string;
+        });
+        const httpParams = new HttpParams().appendAll(this.urlParams);
+        return this.animeService.getPaginationAndAnimeList(httpParams).pipe(
+          map(response => {
+            this.paginationComponent.setLength(response.count);
+            return response.results;
+          }),
+        );
+      }),
+    );
+  }
+
+  /** A lifecycle hook. */
```

Comment:
You could use [@inheritdoc](https://jsdoc.app/tags-inheritdoc.html) jsdoc tag for angular hooks
```suggestion
  /** @inheritdoc */
```